.github/workflows/build_and_run_chain_simulator_and_execute_system_test.yml

```diff
@@ -0,0 +1,336 @@
+name: Chain Simulator Build and Integration Test
+
+on:
+  pull_request:
+    branches:
+      - 'main'
+      - 'master'
+      - 'rc/*'
+  workflow_dispatch:
+  issue_comment:
+    types: [created]
+
+permissions:
+  issues: write
+  pull-requests: write
+  contents: read
+
+jobs:
+  build-and-test:
+    if: |
+      github.event_name == 'pull_request' ||
+      (github.event_name == 'issue_comment' && contains(github.event.comment.body, 'Run Tests:')) ||
+      github.event_name == 'workflow_dispatch'
+
+    strategy:
+      matrix:
+        #TODO Include Macos support later on
+        runs-on: [ubuntu-latest]
+    runs-on: ${{ matrix.runs-on }}
+    env:
+      BRANCH_NAME: ${{ github.head_ref || github.ref_name }}
+      TARGET_BRANCH: ""
+      MX_CHAIN_GO_TARGET_BRANCH: ""
+      MX_CHAIN_SIMULATOR_TARGET_BRANCH: ""
+      MX_CHAIN_TESTING_SUITE_TARGET_BRANCH: ""
+
+    steps:
+      - name: Fetch Latest Comment
+        if: github.event_name != 'issue_comment'
+        uses: actions/github-script@v7
+        id: fetch_comment
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          script: |
+            const comments = await github.rest.issues.listComments({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              issue_number: context.issue.number,
+            });
+
+            // Filter for comments containing "Run Tests:"
+            const latestComment = comments.data.reverse().find(comment => comment.body.includes('Run Tests:'));
+
+            if (latestComment) {
+              core.setOutput('latest_comment', latestComment.body);
+            } else {
+              core.setOutput('latest_comment', '');
+            }
+        env:
+          LATEST_COMMENT: ${{ steps.fetch_comment.outputs.latest_comment }}
+
+      - name: Parse Comment for Branches
```

Comment:
Oh, now I see, some parsing is done, directly on the comment.

How about workflow dispatch with input fields (a bit less code)?

https://github.blog/changelog/2021-11-10-github-actions-input-types-for-manual-workflows/

Reply:
Oh, now I see the discussion below (on GH inputs).